bittide-experiments/src/Bittide/Topology.hs

```diff
@@ -256,52 +256,55 @@ instance FromJSON a => FromJSON (TopologyType IO a) where
    where
     tmm = typeMismatch "Topology" v
 
+newtype FUN a = FUN (forall n . SNat n -> a)
```

Comment:
That's an interesting new requirement. Also means we don't need `ImpredicativeTypes` any more.

Reply:
GHC was no longer happy with `Maybe (forall n . SNat n -> a)`